<html>
<head>
<title>TestNG:  Pow Tests</title>
<link href="../testng.css" rel="stylesheet" type="text/css" />
<link href="../my-testng.css" rel="stylesheet" type="text/css" />

<style type="text/css">
.log { display: none;} 
.stack-trace { display: none;} 
</style>
<script type="text/javascript">
<!--
function flip(e) {
  current = e.style.display;
  if (current == 'block') {
    e.style.display = 'none';
    return 0;
  }
  else {
    e.style.display = 'block';
    return 1;
  }
}

function toggleBox(szDivId, elem, msg1, msg2)
{
  var res = -1;  if (document.getElementById) {
    res = flip(document.getElementById(szDivId));
  }
  else if (document.all) {
    // this is the way old msie versions work
    res = flip(document.all[szDivId]);
  }
  if(elem) {
    if(res == 0) elem.innerHTML = msg1; else elem.innerHTML = msg2;
  }

}

function toggleAllBoxes() {
  if (document.getElementsByTagName) {
    d = document.getElementsByTagName('div');
    for (i = 0; i < d.length; i++) {
      if (d[i].className == 'log') {
        flip(d[i]);
      }
    }
  }
}

// -->
</script>

</head>
<body>
<h2 align='center'>Pow Tests</h2><table border='1' align="center">
<tr>
<td>Tests passed/Failed/Skipped:</td><td>5/5/0</td>
</tr><tr>
<td>Started on:</td><td>Thu Jun 28 19:05:33 MSK 2018</td>
</tr>
<tr><td>Total time:</td><td>0 seconds (0 ms)</td>
</tr><tr>
<td>Included groups:</td><td></td>
</tr><tr>
<td>Excluded groups:</td><td></td>
</tr>
</table><p/>
<small><i>(Hover the method name to see the test class name)</i></small><p/>
<table width='100%' border='1' class='invocation-failed'>
<tr><td colspan='4' align='center'><b>FAILED TESTS</b></td></tr>
<tr><td><b>Test method</b></td>
<td width="30%"><b>Exception</b></td>
<td width="10%"><b>Time (seconds)</b></td>
<td><b>Instance</b></td>
</tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powNegativeTest()'><b>powNegativeTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: -2.2, 2.2</td>
<td><div><pre>org.testng.TestException: 
Method PowCalcTestDouble.powNegativeTest(double, double)[pri:0, instance:com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326] should have thrown an exception of type class java.lang.Exception
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
... Removed 20 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace366803687", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace366803687'><pre>org.testng.TestException: 
Method PowCalcTestDouble.powNegativeTest(double, double)[pri:0, instance:com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326] should have thrown an exception of type class java.lang.Exception
	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:80)
	at org.testng.internal.Invoker.considerExceptions(Invoker.java:1140)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:615)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1204)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
</pre></div></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powNegativeTest()'><b>powNegativeTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: -2.2, -2.2</td>
<td><div><pre>org.testng.TestException: 
Method PowCalcTestDouble.powNegativeTest(double, double)[pri:0, instance:com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326] should have thrown an exception of type class java.lang.Exception
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
... Removed 20 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1033917063", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1033917063'><pre>org.testng.TestException: 
Method PowCalcTestDouble.powNegativeTest(double, double)[pri:0, instance:com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326] should have thrown an exception of type class java.lang.Exception
	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:80)
	at org.testng.internal.Invoker.considerExceptions(Invoker.java:1140)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:615)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1204)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
</pre></div></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest()'><b>powTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: 2.5, 2.5, 9.8821176880262</td>
<td><div><pre>java.lang.AssertionError: Sum operation is incorrect: 2.5^2.5 = 6.25 expected [9.8821176880262] but found [6.25]
	at com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest(PowCalcTestDouble.java:13)
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
... Removed 27 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1677207406", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1677207406'><pre>java.lang.AssertionError: Sum operation is incorrect: 2.5^2.5 = 6.25 expected [9.8821176880262] but found [6.25]
	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.failNotEquals(Assert.java:776)
	at org.testng.Assert.assertEqualsImpl(Assert.java:137)
	at org.testng.Assert.assertEquals(Assert.java:118)
	at com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest(PowCalcTestDouble.java:13)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1204)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
</pre></div></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest()'><b>powTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: 2, 2.5, 5.656854249492381</td>
<td><div><pre>java.lang.AssertionError: Sum operation is incorrect: 2.0^2.5 = 4.0 expected [5.656854249492381] but found [4.0]
	at com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest(PowCalcTestDouble.java:13)
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
... Removed 27 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1979274004", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1979274004'><pre>java.lang.AssertionError: Sum operation is incorrect: 2.0^2.5 = 4.0 expected [5.656854249492381] but found [4.0]
	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.failNotEquals(Assert.java:776)
	at org.testng.Assert.assertEqualsImpl(Assert.java:137)
	at org.testng.Assert.assertEquals(Assert.java:118)
	at com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest(PowCalcTestDouble.java:13)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1204)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
</pre></div></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest()'><b>powTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: 2.2, -2.2, 0.1764696816352779</td>
<td><div><pre>java.lang.AssertionError: Sum operation is incorrect: 2.2^-2.2 = 0.09391435011269719 expected [0.1764696816352779] but found [0.09391435011269719]
	at com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest(PowCalcTestDouble.java:13)
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
... Removed 27 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace101874302", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace101874302'><pre>java.lang.AssertionError: Sum operation is incorrect: 2.2^-2.2 = 0.09391435011269719 expected [0.1764696816352779] but found [0.09391435011269719]
	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.failNotEquals(Assert.java:776)
	at org.testng.Assert.assertEqualsImpl(Assert.java:137)
	at org.testng.Assert.assertEquals(Assert.java:118)
	at com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest(PowCalcTestDouble.java:13)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1204)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at Runners.CalcTestsRunner.main(CalcTestsRunner.java:27)
</pre></div></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
</table><p>
<table width='100%' border='1' class='invocation-passed'>
<tr><td colspan='4' align='center'><b>PASSED TESTS</b></td></tr>
<tr><td><b>Test method</b></td>
<td width="30%"><b>Exception</b></td>
<td width="10%"><b>Time (seconds)</b></td>
<td><b>Instance</b></td>
</tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest()'><b>powTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: 2, 3, 8</td>
<td></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest()'><b>powTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: -2, 3, -8</td>
<td></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest()'><b>powTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: 2, -3, 0.125</td>
<td></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest()'><b>powTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: -2, -3, -0.125</td>
<td></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
<tr>
<td title='com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble.powTest()'><b>powTest</b><br>Test class: com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble<br>Parameters: 2.5, 3, 15.625</td>
<td></td>
<td>0</td>
<td>com.epam.cdp.CalcUnitTests.Tests.PowCalcTestDouble@42039326</td></tr>
</table><p>
</body>
</html>